### PR TITLE
kv: clean unnecessary interface function

### PR DESCRIPTION
**Pull request by AilinKid**
Signed-off-by: ailinkid <314806019@qq.com>

### What problem does this PR solve?

Problem Summary:
clean the unnecessary interface code, since it has been defined in the kv interface

### What is changed and how it works?

What's Changed:
remove the unnecessary interface code in the retriever

### Check List 

Tests 

- [x] No code

### Release note 

```release-note
kv: clean unnecessary interface function.
```

### CODE REVIEW

**Review by ngaut — APPROVED** [review]
LGTM

File: kv/kv.go [hunk 1]
```diff
@@ -92,8 +92,6 @@ var (
 type RetrieverMutator interface {
 	Retriever
 	Mutator
-	// set allowed options of current operation in each TiKV disk usage level.
-	SetDiskFullOpt(level kvrpcpb.DiskFullOpt)
```
Comment (tier-cap) [inline kv/kv.go:96]: If this interface is removed, then there are some other interface need to be removed:
One is MemBuffer::SetDiskFullOpt, Another is client-go MemDB::SetDiskFullOpt.

Comment (AilinKid) [reply kv/kv.go:96]: got, I'll move the goclient later

**Review by djshow832 — APPROVED** [review]

**Pull request merged**